dynd/include/types/type_unpack.hpp

```diff
@@ -0,0 +1,199 @@
+//
+// Copyright (C) 2011-15 DyND Developers
+// BSD 2-Clause License, see LICENSE.txt
+//
+
+#pragma once
+
+#include <Python.h>
+
+#include <dynd/type_registry.hpp>
+#include <type_traits>
+
+#include "pyobject_type.hpp"
+
+using namespace dynd;
+
+
+// Unpack and convert a single element to a PyObject.
+template <typename T>
+inline std::enable_if_t<std::is_signed<T>::value && std::numeric_limits<T>::max() <= INT64_MAX, PyObject *>
+unpack_single(const char *data)
+{
+    return PyLong_FromLongLong(*reinterpret_cast<const T *>(data));
+}
+
+template <typename T>
+inline std::enable_if_t<std::is_unsigned<T>::value && std::numeric_limits<T>::max() <= UINT64_MAX, PyObject *>
+unpack_single(const char *data)
+{
+    return PyLong_FromUnsignedLongLong(*reinterpret_cast<const T *>(data));
+}
+
```

Comment:
Sure, I'll look into it. That bug is too old for it not to be fixed in MSVC 2015. Stuff like this usually comes when one compiler decides to be more or less forgiving about an ambiguous aspect of the standard. MSVC and clang/gcc tend to disagree about when to be strict about the standard and when to let things compile anyway.